lib/rust/parser/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+#![feature(test)]
+#![deny(unconditional_recursion)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unsafe_code)]
+#![warn(unused_import_braces)]
+
+//! This module exports the implementation of parser for the Enso language.
+
+use ast::AnyAst;
+use ast::Ast;
+
+use jni::JNIEnv;
+use jni::objects::*;
+use jni::sys::*;
+
+
+
+// =======================
+// === Parser Rust API ===
+// =======================
+
+/// Parses a single source file.
+pub fn parse(input:String) -> AnyAst {
+    Ast::new(ast::txt::Text{text:input})
+}
```

Comment:
You're missing the additional functions we talked about, such as `tokens` to perform lexing only.